src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/SubjectAccessRequestServiceTest.kt

```diff
@@ -0,0 +1,212 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import org.junit.jupiter.api.Test
+import org.skyscreamer.jsonassert.JSONAssert
+import org.skyscreamer.jsonassert.JSONCompareMode
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1ApplicationTimelinessCategory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.PersonRisksFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.ApprovedPremisesApplicationStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.ApprovedPremisesType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.RiskTier
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.RiskWithStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.subjectaccessrequest.SubjectAccessRequestService
+import java.time.LocalDate
+import java.time.OffsetDateTime
+
+class SubjectAccessRequestServiceTest : IntegrationTestBase() {
+
+  companion object {
+    const val CRN = "CRN123"
+    const val NOMS_NUMBER = "A123456"
+    val START_DATE = java.time.LocalDateTime.of(2018, 9, 30, 0, 0, 0)
+    val END_DATE = java.time.LocalDateTime.of(2024, 9, 30, 0, 0, 0)
+    const val CREATED_AT = "2021-09-25T16:00:00+00:00"
+    const val SUBMITTED_AT = "2021-10-25T16:00:00+00:00"
+    const val ARRIVED_AT = "2021-09-26T16:00:00+00:00"
+
+    const val DATA_JSON_SIMPLE = """{ "key": "value" }"""
+    const val DOCUMENT_JSON_SIMPLE = """{ "key2": "value2" }"""
+    const val RISK_JSON = """"""
+    const val EVENT_NUMBER = "1"
+    const val OFFENCE_ID = "BEING_BAD"
+    const val CONVICTION_ID = 2L
+    const val RELEASE_TYPE_CONDITIONAL = "CONDITIONAL"
+    const val WITHDRAWAL_REASON_NOT_WITHDRAWN = "NOT WITHDRAWN"
+    const val OTHER_WITHDRAWAL_REASON_NOT_APPLICABLE = "NOT APPLICABLE"
+    const val SENTENCE_TYPE_CUSTODIAL = "CUSTODIAL"
+
+    // null test values
+    val CRN_NULL = null
+    val NOMS_NULL = null
+    val START_DATE_NULL = null
+    val END_DATE_NULL = null
```

Comment:
i don't think these NULL values are used?

Reply:
Yep - will clean up on a next pass, was thinking of testing the query.